cognite_toolkit/_cdf_tk/loaders/_resource_loaders/agent_loaders.py

```diff
@@ -0,0 +1,99 @@
+from cognite_toolkit._cdf_tk._parameters.data_classes import ParameterSpec, ParameterSpecSet
+from cognite_toolkit._cdf_tk.client.data_classes.agents import Agent, AgentList, AgentWrite, AgentWriteList
+from cognite_toolkit._cdf_tk.loaders._base_loaders import ResourceLoader
+
+
+class AgentLoader(ResourceLoader[str, AgentWrite, Agent, AgentWriteList, AgentList]):
+    folder_name = "agents"
+    filename_pattern = r".*\.Agent$"  # Matches all yaml files whose stem ends with '.Agent'.
+    resource_cls = Agent
+    resource_write_cls = AgentWrite
+    list_cls = AgentList
+    list_write_cls = AgentWriteList
+    kind = "Agent"
+    _doc_base_url = ""
+    _doc_url = "https://pr-2829.specs.preview.cogniteapp.com/20230101-alpha.json.html"
+
+    @classmethod
+    def get_id(cls, item: AgentWrite | Agent | dict) -> str:
+        if isinstance(item, dict):
+            return item["externalId"]
+        return item.external_id
+
+    @classmethod
+    def dump_id(cls, id: str) -> dict[str, Any]:
+        return {"externalId": id}
+
+    @classmethod
+    def get_required_capability(
+        cls, items: Sequence[AgentWrite] | None, read_only: bool
+    ) -> Capability | list[Capability]:
+        return []
+
+    def load_resource_file(
+        self, filepath: Path, environment_variables: dict[str, str | None] | None = None
+    ) -> list[dict[str, Any]]:
+        loaded = super().load_resource_file(filepath, environment_variables)
+        for item in loaded:
+            # the API always returns instructions
+            if "instructions" not in item:
+                item["instructions"] = ""
+        return loaded
+
+    @classmethod
+    @lru_cache(maxsize=1)
+    def get_write_cls_parameter_spec(cls) -> ParameterSpecSet:
+        spec = super().get_write_cls_parameter_spec()
+        # Tool configuration is a dict where the accepted keys depend on the tool type.
+        spec.add(
+            ParameterSpec(
+                ("tools", ANY_INT, "configuration", ANYTHING),
+                frozenset({"dict"}),
+                is_required=False,
+                _is_nullable=False,
+            )
+        )
+        return spec
+
+    def create(self, items: AgentWriteList) -> AgentList:
+        return self.client.agents.apply(items)
+
+    def retrieve(self, ids: SequenceNotStr[str]) -> AgentList:
+        return self.client.agents.retrieve(ids)
+
+    def update(self, items: AgentWriteList) -> AgentList:
+        return self.client.agents.apply(items)
+
+    def delete(self, ids: SequenceNotStr[str]) -> int:
+        try:
+            self.client.agents.delete(ids)
+            return len(ids)
+        except CogniteAPIError:
+            deleted = 0
+            for id in ids:
+                try:
+                    self.client.agents.delete(id)
+                    deleted += 1
+                except CogniteAPIError:
+                    pass
+            return deleted
```

Comment:
Nitpick, it is best practise to only have the code line(s) that can trigger the exception inside the try... except block
```suggestion
        try:
            self.client.agents.delete(ids)
        except CogniteAPIError:
            deleted = 0
            for id in ids:
                try:
                    self.client.agents.delete(id)
                    deleted += 1
                except CogniteAPIError:
                    pass
            return deleted
        else:
            return len(ids)
```